### DISCUSSION TITLE
词汇表讨论

### DISCUSSION THREAD

**Discussion by antfu** [windicss/docs-cn#8] [General]
请利用讨论区的回复功能进行不同词汇的讨论。

已确定的词汇表：https://github.com/windicss/docs-cn/issues/4

**Comment by antfu** [comment]
## Utilities

直译 `实用程序`

@antfu 提案 `工具类`

Tailwind 中文文档翻译 `功能类` https://www.tailwindcss.cn/

**Reply by KimYangOfCat** [reply]
赞同 `工具类`

**Reply by QC-L** [reply]
工具类

**Comment by antfu** [comment]
## Variants

提案 `变体`

**Reply by HerbertHe** [reply]
这个我在editors/vscode里面使用了 `变量` ，感觉有点不太合适

**Reply by HerbertHe** [reply]
在tailwind的中文文档也是变体，减少tailwind到windi的迁移成本，一致翻译？

**Reply by KimYangOfCat** [reply]
> 这个我在editors/vscode里面使用了 `变量` ，感觉有点不太合适

我查看了editors/vscode，感觉 `变量` 还是蛮适合的唉，个人赞同 `变量`

**Reply by antfu** [reply]
变量 应该是 variables. 在 Windi 里 variant 指通过前缀对工具进行修饰，类似 `font-bold` -> `sm:font-bold`。这里并没有涉及到`量`，翻译成变量感觉会有歧义

**Reply by KimYangOfCat** [reply]
> 变量 应该是 variables. 在 Windi 里 variant 指通过前缀对工具进行修饰，类似 `font-bold` -> `sm:font-bold`。这里并没有涉及到`量`，翻译成变量感觉会有歧义

嗷嗷，明白了，如果是这样的话，确实和现有的 `变量` 一词有歧义。但是关于 `变体` 的，似乎也不是一个人人熟知的概念，建议翻译时，在该词第一次出现的地方写上译者注释，简要介绍该词。例如：

`Intelligent suggestions for utilities and variants.`
可译为：`为工具类和变体（译者注：指通过前缀对工具进行修饰）提供智能建议。`

**Reply by HerbertHe** [reply]
我也是感觉变量是会导致variable的歧义，但是变体的话并不太是一个好理解的词。根据代码的意思的话，感觉 `可变修饰` 好像算是比较符合意思

**Reply by QC-L** [reply]
赞同，意译的话，`可变修饰` 会更好一些，更贴合实际用法

**Reply by KimYangOfCat** [reply]
赞同 `可变修饰` ，`变体` 确实不太好理解

**Reply by ShenQingchuan** [reply]
我赞同 @QC-L 提出的 “ 可变修饰 ” 
但我觉得可以不必要保留 Vari- 词根所带来的那个 “ 变 ” 字的意思

或者说...🤔那种 misunderstood 的感觉来自于咱们这个 “可“ 字，
本身 Variants 这个词并不含这样 ”能够“ 的意思。

对某个工具类的修饰 本质上就是在对其进行改变 不是么？
所以为了中文读者理解更直观，不妨试试采用 `修饰符` 这样的翻译？

**Reply by voorjaar** [reply]
响应类 怎么样？ 类似responsive design的感觉

**Reply by HerbertHe** [reply]
> 我赞同 @QC-L 提出的 “ 可变修饰 ”
> 但我觉得可以不必要保留 Vari- 词根所带来的那个 “ 变 ” 字的意思
> 
> 或者说...🤔那种 misunderstood 的感觉来自于咱们这个 “可“ 字，
> 本身 Variants 这个词并不含这样 ”能够“ 的意思。
> 
> 对某个工具类的修饰 本质上就是在对其进行改变 不是么？
> 所以为了中文读者理解更直观，不妨试试采用 `修饰符` 这样的翻译？

`可变修饰` 是意译翻译，修饰是指在样式前加上标记，在不同状态下的样式表示。可变代表的是变化本身，因为感觉 `变修饰` 一词太奇怪了，修饰的符号把响应式设计和伪类结合在一起了，感觉 `响应标记` 并不是太准确~

**Reply by HerbertHe** [reply]
我觉得 `可` 一字也有意义，使能的意思。如果不加标记的话，所有的样式其实是固定的，而伪类和窗口大小的变化使样式发生了改变了~

**Reply by QC-L** [reply]
`修饰符` 会和 Modifiers 混淆吧？尽量避免混淆...

本身是变体，感觉要突出 `变` 的

**Reply by HerbertHe** [reply]
> `修饰符` 会和 Modifiers 混淆吧？尽量避免混淆...
> 
> 本身是变体，感觉要突出 `变` 的

我感觉修饰就可以了，修饰符突出的是符号，但修饰是这个使能的动作。修饰符可以替换的词有很多，我感觉这个过程更侧重于提供这样的能力，感觉 `可变修饰` 还是合适的

**Reply by QC-L** [reply]
嗯，就 `可变修饰` 就好了

**Comment by KimYangOfCat** [comment]
Windi CSS IntelliSense 
---
现在的翻译为：`Windi CSS 智能感知` #6 
提案：`Windi CSS IntelliSense` 不翻译
这是一个 VS Code 插件，`Windi CSS IntelliSense`  本身就是这个插件的名字，不建议翻译。目前的翻译阅读起来有些奇怪

**Reply by HerbertHe** [reply]
连带着webstorm的翻译已经修正

**Reply by ShenQingchuan** [reply]
Intellisense 我记得有一个译法：智能提示 
更强调给开发者键入代码时的提示

不一定要用 ”感知“ 去确切地对应 sense。

**Reply by voorjaar** [reply]
不翻译可以，翻译的话就智能补全，智能提示，或者vscode插件

**Reply by HerbertHe** [reply]
在PR里面的讨论和实际翻译内容来看，还是不翻译比较好了~ 毕竟CV到vscode找插件也比较方便（不是

**Comment by antfu** [comment]
## Preflights

Related #12, #10

Preflight 类似 normalize.css, 一些基础 CSS 用于重置样式。不确定如何翻译

https://www.tailwindcss.cn/docs/adding-base-styles

**Reply by HerbertHe** [reply]
在svelte集成的部分我翻译为了 `预检查`

preflights 一词对应的应该是飞机起飞前的预检查。通常在飞机起飞前，飞行员需要对着检查单检查是否各部分都正常，与文档中的意思还是比较相符的。

**Reply by guangzan** [reply]
tailwindcss [中文文档 - Preflight](https://www.tailwindcss.cn/docs/preflight) 中没有翻译 **Preflight**，而是给予概念：Preflight 是一套针对 Tailwind 项目的基础样式。翻译成 “预检” 或 “预检查”，侧重 “提前” 的概念，“检查” 比较宽泛。翻译成 “预设样式” 较符合文档表述的语境，易理解、直观一些。

**Reply by voorjaar** [reply]
提议基础样式或者预检样式，我记得中文css经常翻译成样式表，所以可以对应上。

**Reply by HerbertHe** [reply]
从原文档里面看的话，我还是觉得 `预检查` 会好一些，主要是因为在我的 Svelte 翻译部分。preflights 部分提醒的偏重点并不是在样式是啥上面，而是注重于处理其代表样式的作用域的问题。在代码部分 `preflights` 是一个属性标记，如下

```svelte
<!-- use this for scoped preflights -->
<style windi:preflights>
</style>

<!-- use this for global preflights -->
<style windi:preflights:global>
</style>
```

**Reply by HerbertHe** [reply]
如果是样式的话，可能会对 `style` 和 `class` 所表示样式内容产生误解，此部分主要处理的问题是规范化样式的引入过程

**Reply by guangzan** [reply]
`windi:preflights` 这里的 preflights 仍然是指 “预设的样式”，只是通过语法糖把样式限定在 scope。提议：”预检样式“ 或 ”预设样式“，结合你给出的示例，”预检样式“ 更恰当一些。

**Reply by HerbertHe** [reply]
我感觉 `预检样式` 会好一些，预设总让我感觉是 presets

**Reply by QC-L** [reply]
预检样式

**Reply by HerbertHe** [reply]
MDN 上也有这个词 https://developer.mozilla.org/zh-CN/docs/Glossary/Preflight_request ，预检请求给CORS做的检查过程

**Comment by ShenQingchuan** [comment]
## 顶栏 Utilities 中的 Effects

**提案：`影效`**

该章节其中内容主要涵盖：阴影程度、颜色和混合模式

- 影：取自 “影响”，但这里如果和 Vue3 里那样译作 副作用又似乎偏离了此章内容的本义。
- 效：因为内容都是与展示效果有关

**Reply by guangzan** [reply]
中文 “影效” 一般指 “影视特效”，建议直译成 “特效”。 Effects 包含 Box Shadow 、Opacity、Mix Blend Mode。

**Reply by HerbertHe** [reply]
我感觉直接 `效果` 就好了，虽然挺土的不过没啥理解障碍，万一要找文档也好找一些~

**Reply by QC-L** [reply]
赞同 `特效`（特殊效果）

**Reply by ShenQingchuan** [reply]
对对对！！我想起来了😂 做特效的那个软件 ae 就是 安after effects

要不就 特效

**Comment by ShenQingchuan** [comment]
我们是否应该专门先将 顶栏中的所有大类的词先确定了？因为这些词感觉是其相应各个章节中常要用到的词。

**Reply by voorjaar** [reply]
### 提议
通用：颜色/变体
基础：文字/背景/边框/图标
布局：容器/间距/尺寸/显示/排列/网格/表格/定位
增强：滤镜/特效/行为/过渡/转换

**Reply by ShenQingchuan** [reply]
我看 tailwind 中文文档没有翻译 display，我这些比较直接的 css 本来的属性名要不就不翻译了吧

**Comment by HerbertHe** [comment]
### safe list

~~提议 `安全清单`~~

确定为 `白名单`

在好几部分都出现了这个，原文的意思是支持动态工具类的渲染，需要将所有可能的工具类加入safelist，感觉 `安全清单` 还是合适的。

#15 #14 #13 #12

**Reply by antfu** [reply]
本意是白名单，因为之前上升到种族问题所以大部分软件都避开使用 whitelist/blacklist。对应 blocklist 为黑名单。

**Reply by QC-L** [reply]
要不要改成 dennylist/allowlist

**Reply by antfu** [reply]
嗯？

**Reply by QC-L** [reply]
禁止/允许

**Reply by HerbertHe** [reply]
感觉直译会好一些，安全清单没有给使用带来误解，而且确实是枚举之后才可以使用。

在配置项也是用了safelist，直译的话对配置理解还是挺友好的。

**Reply by KimYangOfCat** [reply]
虽然直译的 `安全清单` 也并没有歧义，但是这种译法当前比较少见， `白名单` 这种译法存在的时间更长，对于大家而言更熟悉，所以我赞同 `白名单`

**Reply by HerbertHe** [reply]
我怕白名单总会想到 whitelist ，而配置项是 safelist

**Reply by yangchaojun** [reply]
`白名单`的翻译更加符合常识

**Reply by KimYangOfCat** [reply]
> 我怕白名单总会想到 whitelist ，而配置项是 safelist

在英文中为避免种族问题才避开 whitelist 这个词的，但是在中文中，`白名单` 这个词并不涉及任何种族问题，所以我认为也没有避开的必要。

**Reply by HerbertHe** [reply]
> > 我怕白名单总会想到 whitelist ，而配置项是 safelist
> 
> 在英文中为避免种族问题才避开 whitelist 这个词的，但是在中文中，`白名单` 这个词并不涉及任何种族问题，所以我认为也没有避开的必要。

倒不是怕种族注意的问题，emmmm，怕配置的时候写错，虽然注意一下就可以了🌚

**Reply by KimYangOfCat** [reply]
根据目前的讨论，`白名单` 有 3 人支持， `安全清单`有 1 人，要不就定 `白名单`了？ @HerbertHe

**Reply by HerbertHe** [reply]
> 根据目前的讨论，`白名单` 有 3 人支持， `安全清单`有 1 人，要不就定 `白名单`了？ @HerbertHe 

可以的，我把之前合并过的翻译改一下～

**Comment by HerbertHe** [comment]
## on-demanded

提议 `随需启用`

> Preflight is enables on demanded, if you'd like to completely disable it, you can configure it as below

在Rollup的翻译中，`preflight` 本身在使用到的情况下才会开启。我感觉 `随需启用` 比 `按需启用` 更能表达原本的含义，这部分的内容是关于直接完全关掉 preflight 的

**Reply by KimYangOfCat** [reply]
赞同原本的 `按需启用`

**Comment by antfu** [comment]
## Light Mode / Dark Mode

亮色模式 / 暗色模式

#19

**Comment by antfu** [comment]
## Explicit / Implicit

显式 / 隐式

#18

**Comment by HerbertHe** [comment]
# Important prefix

提议 `important 前缀`

#23 

如果翻译为 `重要的前缀` 的话，并不利于理解。保留 `important` 不翻译，更好地联想到 `!important` 的前缀标记

**Reply by KimYangOfCat** [reply]
赞同楼上

**Comment by HerbertHe** [comment]
# Grid Templates

提议不翻译

原文在CSS中有直接的对应部分，翻译之后可能会造成困扰

#28

**Reply by KimYangOfCat** [reply]
可否列出相关的PR？方便大家查看原文

**Comment by HerbertHe** [comment]
# opt-out

不知道如何翻译

#28

**Reply by KimYangOfCat** [reply]
可否列出相关的PR？方便大家查看原文

**Reply by HerbertHe** [reply]
> 可否列出相关的PR？方便大家查看原文

列了列了

**Comment by Megrax** [comment]
## CSS Escape
不知道该怎么翻译...或许不需要翻译？
#29

**Reply by HerbertHe** [reply]
`CSS 转义`

**Comment by KimYangOfCat** [comment]
scoped styles 
---
提议： 不翻译
相关PR：https://github.com/windicss/docs-cn/pull/10#pullrequestreview-668701344 #12

**Reply by HerbertHe** [reply]
这个在svelte中是可以翻译的，默认情况下svelte的样式全都是受作用域限制的。

**Reply by HerbertHe** [reply]
svelte中并没有特殊的scope的标记

**Comment by antfu** [comment]
## Class (CSS)

`类`？

不翻译感觉又很奇怪

#28 

```diff
Since Windi CSS will only generate the CSS utilities you use, it enables the possibility to use arbitrary value in your classes and generate coresponding styles based on the appropriate semantics.
-Windi CSS 只会生成你使用的 CSS 工具类，它允许你在 classes 中写任意的值，基于适当的语义生成相应的样式。
+Windi CSS 只会生成你使用的 CSS 工具类，它允许你在类名中写任意的值，基于适当的语义生成相应的样式。
```

**Reply by QC-L** [reply]
类

**Reply by HerbertHe** [reply]
我感觉 class 或者 className 不翻译更好一些，`类` 感觉联想 class 还有点障碍。不过见到 `class` ，前端基本上都知道是指的啥

**Reply by KimYangOfCat** [reply]
我支持翻译为 `类`，至于楼上同学说的联想障碍，我个人认为应该不会吧，这是一个 CSS 框架的文档，一提类，应该不会错误联想到面向对象的`类`上吧。

**Comment by KimYangOfCat** [comment]
purge-based
----
不知如何翻译，目前翻译为：基于清除的
相关PR：#21

**Reply by antfu** [reply]
`基于 Purge` 的就好，这个词是在映射 Purge CSS

**Comment by HerbertHe** [comment]
## Stroke

提议与之相关的词语组合都不翻译，在 MDN 上跟 SVG 有关的词组没有翻译，不翻译也不影响理解。如果翻译的话，可能会适得其反。

#51

**Comment by luckyJeffy** [comment]
## Ring

直译 `环`
Tailwind 中文文档翻译 `轮廓环` https://www.tailwindcss.cn/docs/ring-width
提议 `环绕轮廓`

相关PR https://github.com/windicss/docs-cn/pull/54

**Reply by HerbertHe** [reply]
根据上下文，我觉得 `轮廓环` 就已经很准确表达 `outline ring` 的意思了。环绕一词为动词，而文档中表达为名词。

**Comment by ShenQingchuan** [comment]
对现行版本的 features/shortcuts 的标题存有疑虑：

shortcuts：根据苹果词典

shortcut | BrE ˈʃɔːtkʌt, AmE ˈʃɔrtˌkət |
  noun
  ① (route) 近路 jìnlù
      ▸ to take a shortcut through the park
      穿过公园走近路
  ② (method) 捷径 jiéjìng
      ▸ to take shortcuts
        informal 走捷径
      ▸ there are no shortcuts to becoming a musician
        想要成为音乐家是无捷径可走的
  ③ Computing 快捷键 kuàijiéjiàn
      ▸ a keyboard shortcut
        快捷键 kuàijiéjiàn
      ▸ a shortcut key
         键盘快捷键
_________________________

我认为：翻译为 “ 捷径 ” 更符合实际意义。

**Reply by HerbertHe** [reply]
我感觉不管是现在的快捷方式的翻译还是捷径，都不太合适。此处我觉得意译可能会更好一点，以降低用户的心智负担。

根据shortcuts部分实际表达的意思，其实应该是指 custom utilities collection，跟在样式表里面使用 `@apply` 没什么区别，只是样式表更具有拓展性。

因此，我提议 `工具集` 可能会更好降低心智负担。

**Reply by antfu** [reply]
建议保留英文原文，以便和配置项对应避免疑惑

**Comment by HerbertHe** [comment]
# Font family

提议不翻译

#66

**Comment by HerbertHe** [comment]
# Flexbox 及其相关 CSS 属性名

提议不翻译

#68

**Comment by watonyweng** [comment]
- AutoEscape

> 提议： 自动转义
> 相关PR：#72

**Comment by watonyweng** [comment]
- Mix Blend Mode

> 提议： 不翻译
> 相关PR：#74
> 来源：[Mix Blend Mode](https://developer.mozilla.org/zh-CN/docs/Web/CSS/mix-blend-mode)

**Comment by watonyweng** [comment]
- Perspective/Perspective Origin

> 提议： 不翻译
> 相关PR：#76
> 来源：
>  - [perspective](https://developer.mozilla.org/zh-CN/docs/Web/CSS/perspective) 
>  - [perspective-origin](https://developer.mozilla.org/zh-CN/docs/Web/CSS/perspective-origin)

**Comment by HerbertHe** [comment]
## inline

提议：`行内`

翻译参考MDN

#68

**Comment by HerbertHe** [comment]
## Scroll Snap

提议不翻译

MDN 相关 [scroll-snap-type](https://developer.mozilla.org/zh-CN/docs/Web/CSS/scroll-snap-type)

PR: #111
